### PR TITLE
Add chromap as aligner

### LINKED ISSUE #233: Add CHROMAP as aligner
From CHROMAP [pre-print](https://www.biorxiv.org/content/10.1101/2021.06.18.448995v1) it is stated that [chromap](https://github.com/haowenz/chromap) performs similarly in terms of accuracy to `bwa` while it is 10X faster. Might be interesting to enable it use.

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Add `chromap` as one of the possible aligners of the pipeline.
Should at some point the chromap indexed add to the igenomes @drpatelh ?

Closes #233 

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] `CHANGELOG.md` is updated

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -221,6 +221,32 @@ if (params.aligner == 'bowtie2') {
...
+                path: { "${params.outdir}/genome/${params.aligner}/index" },
+                mode: 'copy',
+                saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+            ]
+        }
+        withName: CHROMAP_CHROMAP {
+            ext.args   = [
+                            '--preset chip',
+                            '--SAM'
+                        ].join(' ').trim()
```
Comment (drpatelh) [inline conf/modules.config:239]:
```suggestion
            ext.args   = '--preset chip --SAM'
```

File: subworkflows/nf-core/align_chromap.nf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+include { CHROMAP_CHROMAP   } from '../../modules/nf-core/modules/chromap/chromap/main'
+include { BAM_SORT_SAMTOOLS } from './bam_sort_samtools'
+
+workflow ALIGN_CHROMAP {
+    take:
+    reads         // channel: [ val(meta), [ reads ] ]
+    index         //    path: /path/to/index
+    fasta         //    path: /path/to/fasta
```
Comment (drpatelh) [inline subworkflows/nf-core/align_chromap.nf:12]:
```suggestion
    reads // channel: [ val(meta), [ reads ] ]
    index //    path: /path/to/index
    fasta //    path: /path/to/fasta
```

**Review by drpatelh — APPROVED** [review]
Looks awesome 😍 Maybe update the CHANGELOG too?

**Comment by JoseEspinosa** [comment]
> Looks awesome heart_eyes Maybe update the CHANGELOG too?

Was updated already for this PR [here](https://github.com/nf-core/chipseq/pull/267/commits/3fc73c3828846efb769dab288f17af56dbee563c), although probably not for all the previous one :eyes:

**Comment by drpatelh** [comment]
PS: No need to add indices for Chromap to iGenomes. They can be maintained locally by users.

**Pull request merged**